### PR TITLE
Make macOS use libcurl+OpenSSL in Travis CI

### LINKED ISSUE #4650: Client Certificate Authentication in WebCmdlets Unavailable for macOS
Due to an upstream incompatibility between CoreFX  and macOS's implementation of curl (dotnet/corefx#19718), the addition of Client Certificate Authentication in PowerShell 6 (#4546) presents the user with an error on macOS when attempting to use `-Certificate` with `Invoke-Webrequest` and `Invoke-RestMethod`

Steps to reproduce
------------------

```powershell
$certficiate = Get-PFXCertificate -Path c:\path\to\cert.pfx
$Result = Invoke-WebRequest -uri https://localhost:8083 -Certificate $certificate
$Result.StatusCode
```

Expected behavior
-----------------

```none
200
```

Actual behavior
---------------

An error message containing the following:
```none
The handler does not support client authentication certificates with this combination of libcurl (7.55.1)
```

Environment data
----------------

```none
Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.5-78-gcdbbb7a4bb51929470a5ace096999db155a865b2
```
on macOS 10.12

References
-----------------
#3648

### LINKED ISSUE #5590: Tests related to `-SslProtocol` in web cmdlets are failing on macOS daily runs
Same failures from 2 macOS daily builds. And we got the similar failures for `Invoke-RestMethod -SslProtocol`.
```
    [-] Verifies Invoke-WebRequest -SslProtocol Tls11 works on Tls11 145ms
      SSL connect error
      at line: 1527 in /Users/travis/build/PowerShell/PowerShell/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1
    [-] Verifies Invoke-WebRequest -SslProtocol Tls12 works on Tls12 125ms
      SSL connect error
      at line: 1527 in /Users/travis/build/PowerShell/PowerShell/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1
+++
    [-] Verifies Invoke-WebRequest -SslProtocol -SslProtocol Tls11 fails on a Tls only connection 213ms
      Expected string length 83 but was 13. Strings differ at index 0.
      Expected: {WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand}
      But was:  {No Exception!}
      -----------^
      at line: 75 in /Users/travis/build/PowerShell/PowerShell/test/tools/Modules/HelpersCommon/HelpersCommon.psm1
      75:             $_.FullyQualifiedErrorId | Should Be $FullyQualifiedErrorId | Out-Null
    [-] Verifies Invoke-WebRequest -SslProtocol -SslProtocol Tls12 fails on a Tls only connection 181ms
      Expected string length 83 but was 13. Strings differ at index 0.
      Expected: {WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand}
      But was:  {No Exception!}
      -----------^
      at line: 75 in /Users/travis/build/PowerShell/PowerShell/test/tools/Modules/HelpersCommon/HelpersCommon.psm1
      75:             $_.FullyQualifiedErrorId | Should Be $FullyQualifiedErrorId | Out-Null
```

### PR DESCRIPTION
**Pull request by markekraus**
Closes #5590
Closes #4650 

macOS in Travis CI was always using the system provided libcurl instead of the libcurl+openssl installed through `brew`. This resulted in many false fails and features to "not work". This PR makes `pwsh` use libcurl+openSSL and reenables all of the Web Cmdlets tests that were set to pending.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
       rvm install ruby-2.3.3;
       rvm --default use 2.3.3;
+      export DYLD_LIBRARY_PATH=/usr/local/opt/curl/lib:${DYLD_LIBRARY_PATH};
```
Comment (daxian-dbw) [inline .travis.yml]: Can you add a comment to explain why this line is added? It would make the future maintenance easier.

**Review by daxian-dbw — APPROVED** [review]
We need to document the workaround of `export DYLD_LIBRARY_PATH=/usr/local/opt/curl/lib:${DYLD_LIBRARY_PATH};` somewhere for users to enable this on OSX 10.12.6.

File: .travis.yml (hunk 1 above)
Comment (markekraus) [reply .travis.yml]: Sure. I will add it after the tests run. I may need to set some back to pending.

**Comment by markekraus** [comment]
@daxian-dbw actually.. it doesn't completely work with macOS 10.13.1. some of the features still don't work right with libcurl+LibreSSL.

**Comment by TravisEz13** [comment]
@markekraus , we should document the requirements whatever they are.

**Comment by daxian-dbw** [comment]
@markekraus Thanks for clarifying that. I have removed the note about macOS 10.13.1.

**Comment by markekraus** [comment]
PowerShellGet has a line in `Ping-Endpoint` that sets `HttpClientHandler.UseDefaultCredentials` to true. apparently... the libcurl+openssl on macOS doesn't support that feature.

I have no clue about the other fails in the web cmdlets tests... could be bad bad bad coincidence. They passed in my branch earlier. I'll need to do more testing.

But, unless a change is made to PowerShellGet is updated.. this PR can't be merged. 

I'll close this PR and revisit it later.

**Pull request closed without merging**